### PR TITLE
Move Service and Context to their own package.

### PR DESCRIPTION
**Pull request by vdemeester**
Continuing to split the `docker` package in several smaller packages, this time `service` and `ctx` 🐮.

- `service` package holds service related struct and methods. It's just
  a move, stuff like `Namer` should be moved away from this package and
  refactor in a way they don't call the docker API (e.g).
- `ctx` holds the `Context` struct used to pass factory and stuff :)

/cc @joshwget

🐸
Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: project/project.go [hunk 1]
```diff
@@ -15,6 +15,9 @@ import (
 	"github.com/docker/libcompose/yaml"
 )
 
+// ComposeVersion is name of docker-compose.yml file syntax supported version
+const ComposeVersion = "1.5.0"
```
Comment (vdemeester) [inline project/project.go]: We might want to *up* that sometimes as we support compose file v2 which is... way older than `1.5.0`👼

Comment (joshwget) [inline project/project.go]: If we switch this to "2", does that just indicate that we have support for v2 files? If so, I think it shouldn't hurt to bump this now.

Comment (vdemeester) [inline project/project.go]: it's based on the compose version (so something like `1.7.0`, …)

Comment (vdemeester) [inline project/project.go]: anyway, that's another *story* than this PR :stuck_out_tongue_closed_eyes:

**Comment by joshwget** [comment]
LGTM

**Pull request merged**